Create `failed`, `deleted`, `is_green`, `is_yellow` flags in new_run.

vdbergh opened the pull request:
Queries in mongodb for missing flags are a bit messy. This is the first step at fixing this. The next step is to loop through the runs collection and set/unset these flags as needed. **Both steps should be mostly non functional for now** (except for a few bug fixes along the way).

Also: we document where the flags

`finished`, `failed`, `deleted`, `is_green`, `is_yellow` are set and unset.

ppigazzini commented:
DEV updated

ppigazzini commented:
It's a functional (and an enhancement one). Before this PR a "failed" run can be restarted with "modify" but it was still flagged as "failed" after stopping.
I reverted (#994) a my commit  for this issue.

vdbergh commented:
> It's a functional (and an enhancement one) PR. Before this PR a "failed" run can be restarted with "modify" but it was still flagged as "failed" after stopping. I reverted (#994) a my commit for this issue.

I updated to description...

We should have some kind of voting system for stopping a run. E.g. bench failure 3 votes. Make failure 1 vote. Stop the run after 10 votes.

ppigazzini commented:
PROD updated, thank you @vdbergh :)
